req2flatpak.py

```diff
@@ -457,7 +458,8 @@ def _query_from_cache(url) -> Optional[str]:
                 return None
 
         def _query_from_pypi(url) -> str:
-            with urllib.request.urlopen(url) as response:
+            # this an internal method that will be called with a trusted URL
+            with urllib.request.urlopen(url) as response:  # nosec
```

Comment:
a more specific ignore rule would be better (to avoid masking of other security problems that were maybe not considered). in this case: `#nosec B310` should do the job.

Reply:
or even better: instead of just silencing the warning, we could add a guard to ensure the url starts with `https`.

Reply:
> a more specific ignore rule would be better (to avoid masking of other security problems that were maybe not considered). in this case: #nosec B310 should do the job.

Nope. Bandit only understands `# nosec`.

Reply:
> in this case: #nosec B310 should do the job.

That works indeed.